api/internal/handler/data_loader/route_export.go

```diff
@@ -0,0 +1,478 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package data_loader
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/apisix/manager-api/internal/core/entity"
+	"github.com/apisix/manager-api/internal/core/store"
+	"github.com/apisix/manager-api/internal/handler"
+	"github.com/apisix/manager-api/internal/log"
+	"github.com/apisix/manager-api/internal/utils"
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/gin-gonic/gin"
+	"github.com/shiningrush/droplet"
+	"github.com/shiningrush/droplet/data"
+	"github.com/shiningrush/droplet/wrapper"
+	wgin "github.com/shiningrush/droplet/wrapper/gin"
+)
+
+type Handler struct {
+	routeStore    store.Interface
+	upstreamStore store.Interface
+	serviceStore  store.Interface
+	consumerStore store.Interface
+}
+
+func NewHandler() (handler.RouteRegister, error) {
+	return &Handler{
+		routeStore:    store.GetStore(store.HubKeyRoute),
+		upstreamStore: store.GetStore(store.HubKeyUpstream),
+		serviceStore:  store.GetStore(store.HubKeyService),
+		consumerStore: store.GetStore(store.HubKeyConsumer),
+	}, nil
+}
+
+func (h *Handler) ApplyRoute(r *gin.Engine) {
+	r.POST("/apisix/admin/routes/export/:ids", wgin.Wraps(h.ExportRoutes,
+		wrapper.InputType(reflect.TypeOf(ExportInput{}))))
+	r.GET("/apisix/admin/exportall/routes", wgin.Wraps(h.ExportAllRoutes))
```

Comment:
Question:
1. Why the first API's method is `POST`
2. "/apisix/admin/exportall/routes" is not very good. cc @liuxiran

Reply:
> Question:
> 
> 1. Why the first API's method is `POST`
> 2. "/apisix/admin/exportall/routes" is not very good. cc @liuxiran

1. `GET` would be better
2. In fact, we didn't find a very satisfactory name either, do you have any suggestions? @starsz

Reply:
GET `/apisix/admin/export/routes/:ids`
GET `/apisix/admin/export/routes`